### ISSUE TITLE
Alternative terms display

### ISSUE THREAD

**Issue by Mrweak** [NatLibFi/Skosmos#614]
Hi~ I have successfully installed skosmos in my machine according to the instructions, but I met a little configuration problem. In my local server, the vocabulary is not marked with "ENTRY TERM" information, which can be shown in the demo website http://skosmos.dev.finto.fi/yso/en/.
I want the my thesaurus display "ENTRY TERM" information(skos:altLabel data), and the alternative words are shown as italic and an arrow just like "abatement of debts → debt provable" in left list. I have tried some configuration methods but did not work. How can I fix it?  Thanks a lot.

**Comment by osma** [comment]
Hi @Mrweak, thanks for trying out Skosmos!
To be able to help, I need a little more information.

Did you try running Skosmos on your own machine with the two example vocabularies (STW Thesaurus and UNESCO Thesaurus) mentioned in the [install tutorial](https://github.com/NatLibFi/Skosmos/wiki/InstallTutorial)? If you did, did the altLabels work in these vocabularies?

What does your vocabulary data look like? I assume you have skos:altLabel triples there. Are the values language tagged?

There is no specific configuration setting for altLabels, they should be displayed automatically if they exist in the data and are properly language tagged.

**Comment by Mrweak** [comment]
@osma Thank you very much for your reply!
I have tried STW Thesaurus and UNESCO Thesaurus in my machine, and the altLabels did not work either. (The altLabels and the prefLabeis are similarly displayed in the box, and there is no display of italic and arrows). 
I have thought about it carefully. When I installed skosmos, the process of configuration FusekiJenaText encountered some problems, so I directly skip this step. Is it a problem here? Does FusekiJenaText only use for faster text search or also help show altLabels?

**Comment by osma** [comment]
The use of jena-text is only relevant for faster text search. Also the alphabetical index relies on the text index, if available. But if you configure Skosmos so it doesn't rely on the text index (i.e. set SPARQL dialect "Generic" in config.inc) then all the functionality should still work, just a bit slower.

The problem is a bit strange, particularly if it happens also with STW and UNESCO Thesaurus. Can you post
- your Fuseki version
- your Skosmos version
- your config.inc and vocabularies.ttl configuration files
- a screenshot showing how labels are shown

**Comment by Mrweak** [comment]
@osma 
fuseki-2.6.0. 
Skosmos 1.9
And I use  XAMPP5.6.30 to build Apache + php.
Both Ubuntu 15.04 and windows 10  have encountered this problem.
Many thanks.
![ab](https://user-images.githubusercontent.com/16514148/27734485-8f58e4e2-5dcd-11e7-94e2-e332dc47653f.png)
[Desktop.zip](https://github.com/NatLibFi/Skosmos/files/1114992/Desktop.zip)

**Comment by osma** [comment]
@Mrweak Thanks for the details and the screenshot. I still don't quite understand what's happening here, so I have to ask a few more questions.
* Are the altLabels displayed on a concept page? For example if you click on Abstracts from the alphabetical index in the display you posted, do you see "ENTRY TERM: Abstract journals" on the page?
* Are the altLabels searchable? If you search for "Aboriginals", do you get any results? (normally you would find "Indigenous peoples")
* You seem to be using Chinese UI language, does it make a difference if you switch to English UI language?
* If you switch to another content language (e.g. French in UNESCO Thesaurus), does it show altLabels?

**Comment by osma** [comment]
Thanks a lot for the details. I think there could be two unrelated issues here:
1. The alphabetical index display is not working. It could be a bug that affects only the Generic SPARQL flavor (when not using jena-text). I will investigate this, but meanwhile I suggest that you try to set up a jena-text index. It should be pretty well explained in the [install tutorial](https://github.com/NatLibFi/Skosmos/wiki/InstallTutorial).
1. The "entry term" field is not displayed when using Chinese UI language. This is probably because the Chinese translation is incomplete, as it has not been kept up to date w.r.t. changes in the codebase. For now I suggest you use the English  UI. If you want to improve the Chinese UI, you can contribute newer translations via the Transifex platform, see the wiki page [Translation](https://github.com/NatLibFi/Skosmos/wiki/Translation) for details.

**Comment by Mrweak** [comment]
@osma 

- I will install Jena-text and try other possible solutions.

- If I solve this problem, I will inform you immediately.

- Thanks again for your patience!

**Comment by Mrweak** [comment]
@osma 
I have solved this problem.
When I set up a jena-text index successfully, the altLabels were displayed properly.
I am very sure that the display of the altLabes is related to jena-text index. In fact, only after setting the jena-text index and switch DEFAULT_SPARQL_DIALECT to JenaText, the altLabels could be displayed in the left vocabulary list.(at least on my computer).
Many thx.

**Comment by osma** [comment]
Glad that you got it working. I created another issue (#615) to track the problem with the alphabetical index. I will close this one as the other problems seem to be sorted out now.

**Comment by osma** [comment]
The alphabetical index issue is now fixed too, thanks @Mrweak for reporting!
